datastore/postgres/indexfiles.go

```diff
@@ -0,0 +1,130 @@
+package postgres
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/indexer"
+	"github.com/quay/claircore/pkg/microbatch"
+)
+
+var (
+	indexFilesCounter = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "claircore",
+			Subsystem: "indexer",
+			Name:      "indexfiles_total",
+			Help:      "Total number of database queries issued in the IndexFiles method.",
+		},
+		[]string{"query"},
+	)
+
+	indexFilesDuration = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: "claircore",
+			Subsystem: "indexer",
+			Name:      "indexfiles_duration_seconds",
+			Help:      "The duration of all queries issued in the IndexFiles method",
+		},
+		[]string{"query"},
+	)
+)
+
+func (s *IndexerStore) IndexFiles(ctx context.Context, files []claircore.File, layer *claircore.Layer, scnr indexer.VersionedScanner) error {
+	const (
+		insert = `
+		INSERT INTO file
+			(path, kind)
+		VALUES
+			($1, $2)
+		ON CONFLICT (path, kind) DO NOTHING;
+		`
+
+		insertWith = `
+		INSERT
+		INTO file_scanartifact (layer_id, file_id, scanner_id)
+		VALUES (
+			(SELECT id FROM layer WHERE layer.hash = $1),
+			(SELECT id FROM file WHERE file.path = $2 AND file.kind = $3),
+			(SELECT id FROM scanner WHERE scanner.name = $4 AND scanner.version = $5 AND scanner.kind = $6)
```

Comment:
I dunno if having 3 subselects on every insert is good for perf

Reply:
Updated to look up the scanner ID and layer ID first